### PR TITLE
Support containers with show conditions

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [x] documentation updated

**Squash commit message**

* Support containers with show conditions
* Automatically switch child if current no longer considered visible
* If no visible child remove current subscription
* Update `containertab` random to only pick among visible children
* error on userdefined containertab switch to non-visible children

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

File: session/containerhandler.go [hunk 1]
```diff
@@ -39,6 +46,8 @@ type (
 		IsMaster          bool               `json:"isMaster"`
 		ExternalReference *ContainerExternal `json:"externalReference"`
 		Type              string             `json:"type"`
+		// Condition fulfilled  show condition. Value is nil on no show condition, true if it should show and false otherwise
+		Condition *string `json:"condition"`
```
Comment (atluq) [inline session/containerhandler.go:50]: this physically hurts🩹

**Review by atluq — COMMENTED** [review]
Looks good! I guess docs need an update, so will have another look then.

What is the reasoning behind picking from visible only for the random mode of the action?

File: session/containerhandler.go (hunk 1 above)
Comment (DnlLrssn) [reply session/containerhandler.go:50]: > 
> 
> Looks good! I guess docs need an update, so will have another look then.
> 
> What is the reasoning behind picking from visible only for the random mode of the action?

user can only see the visible, there's no tab showing for visualizations not visible, so reason is to chose only among those tabs a user would use.

Don't think docs need an update, maybe some clarification?

Comment (DnlLrssn) [reply session/containerhandler.go:50]: you mean the tristate, or the fact they've chosen to send a string?

Comment (DnlLrssn) [reply session/containerhandler.go:50]: should probably make the comment "True" and "False", will update that

Comment (atluq) [reply session/containerhandler.go:50]: Maybe throw in a "visible" [here](https://github.com/qlik-oss/gopherciser/blob/80bb4fc10a278d1f8b85431959c11c17ffb1bb60/generatedocs/data/params.json#L225).

I meant why we dont chose from visible in all modes. But it makes sense to leave it up to the script author, when the mode is not random.

Comment (DnlLrssn) [reply session/containerhandler.go:50]: sure can clarify that part. But as a user I would assume "random" only select a selectable tab.

Comment (atluq) [reply session/containerhandler.go:50]: Meant that it is a string in the message.

**Review by atluq — APPROVED** [review]

File: session/containerhandler.go (hunk 1 above)
Comment (DnlLrssn) [reply session/containerhandler.go:50]: > 
> 
> Maybe throw in a "visible" [here](https://github.com/qlik-oss/gopherciser/blob/80bb4fc10a278d1f8b85431959c11c17ffb1bb60/generatedocs/data/params.json#L225).
> 
done. now I can check the doc checkbox too ;)

Comment (atluq) [reply session/containerhandler.go:50]: 😆

**Pull request merged**